tests/integration/utils.sh

```diff
@@ -0,0 +1,168 @@
+#!/bin/bash
+
+include './src/lib/kwio.sh'
+
+declare -g CONTAINER_DIR # has the container files to build the container images
+declare -g SAMPLES_DIR   # has sample files used accross the integration tests
+declare -g KWROOT_DIR    # local kw dir to be copied to and installed in the containers
+declare -g DISTROS       # distributions we will run the integration tests
+
+# ensure path to directories is absolute
+script_dir=$(realpath "$(dirname "${0}")")
+CONTAINER_DIR="${script_dir}/podman"
+SAMPLES_DIR="${script_dir}/samples"
+KWROOT_DIR=$(realpath "${script_dir}/../..")
+
+# supported distros
+DISTROS=('archlinux' 'debian' 'fedora')
+
+# Builds a container image for the given distro
+function build_distro_image()
+{
+  local distro="${1}"
+  local file="${CONTAINER_DIR}/Containerfile_${distro}"
+
+  podman image build --file "$file" --tag "kw-${distro}" 2> /dev/null
+
+  # Check if the command failed
+  if [[ $? -ne 0 ]]; then
+    fail "($LINENO): Error building the image for distribution ${distro}"
+  fi
+}
+
+# Builds container images and create containers used accross the tests.
+function setup_container_environment()
```

Comment:
Instead of adding a spinner, I guess we should show more relevant information, such as:

```
Setting up environment for integration tests.
[1/6] Building ArchLinux image...
[2/6] Creating ArchLinux container...
[3/6] Building Debian image...
[4/6] Creating Debian container...
[5/6] Building Debian image...
[6/6] Creating Debian container...
Environment for integration tests has been set up.
```

Later on:

```
Tearing down environment for integration tests.
[1/3] Removing ArchLinux container...
[2/3] Removing Debian container...
[3/3] Removing Fedora container...
```

What do you think?

I prefer this way because (1) it is more informative and (2) I prefer not implement shell spinners (I have never done it..).

Reply:
I agree with you :)